app/controllers/api/v1/providers_controller.rb

```diff
@@ -2,8 +2,24 @@ module Api
   module V1
     class ProvidersController < ApplicationController
       def index
-        @providers = Provider.changed_since(params[:changed_since])
-        paginate json: @providers
+        page_size = params[:per_page] || 100
+        @providers = Provider.changed_since(params[:changed_since]).limit(page_size)
+        last_provider = @providers.last
+
+        response.headers['Link'] = if last_provider
```

Comment:
I remembered something over the weekend. There's an edge case to be aware of, and that we might want to code for because it's likely to be a subtle and hard-to-diagnose bug. If there's a transaction in-flight when this request fires off, I think there's a chance that we will miss it here unless we acquire a write-lock or find some other way of mitigating it.

Let's just have a quick chat about this in-person, I'm not 100% sure how to put it down here.

Reply:
We've discussed this in person and have resolved to:

- Get a write-lock before performing this database fetch
- Document the transaction edge case we're dealing with here through a code comment, for posterity